domain/src/main/java/com/globant/domain/usecase/implementation/GetMemeByIdUseCaseImpl.kt

```diff
@@ -1,8 +1,18 @@
 package com.globant.domain.usecase.implementation
 
-import com.globant.domain.repository.MemeRepository
+import com.globant.domain.entity.MemeDetail
+import com.globant.domain.service.MemeService
 import com.globant.domain.usecase.GetMemeByIdUseCase
+import com.globant.domain.util.NOT_FOUND
+import com.globant.domain.util.Result
 
-class GetMemeByIdUseCaseImpl(private val memeRepository: MemeRepository): GetMemeByIdUseCase {
-    override fun invoke(memeId: Int) = memeRepository.getMemeById(memeId)
+class GetMemeByIdUseCaseImpl(private val memeService: MemeService) : GetMemeByIdUseCase {
+    override fun invoke(memeId: Int): Result<MemeDetail> {
+        memeService.getMemeByIdFromApi(memeId).let {
+            return when (it) {
```

Comment:
the when must be in the serviceImpl

Reply:
here you will just do something like

override fun invoke(memeId: Int): Result<MemeDetail> = memeService.getMemeByIdFromApi(memeId)

Reply:
You are right, refactored!